src/modules/integrations/starship.nix

```diff
@@ -0,0 +1,45 @@
+{ pkgs, config, lib, ... }:
+
+{
+  options.starship = {
+    enable = lib.mkEnableOption "Enable the Starship command prompt.";
+
+    package = lib.mkOption {
+      type = lib.types.package;
+      default = pkgs.starship;
+      defaultText = "pkgs.starship";
+      description = "The Starship package to use.";
+    };
+
+    config.enable = lib.mkEnableOption "Enable Starship config override.";
```

Comment:
Maybe manage the starship config also here :)

Reply:
If you mean adding there individual entries to write the file for the user, I purposefully left this to be handled by Starship itself because:

- I don't want to remake work that was already done by [home-manager](https://github.com/nix-community/home-manager/blob/7ae7250df8f38c4efc8cd6669a36272ab7a575ed/modules/programs/starship.nix). I need to research more about how to reuse modules in the whole Nix ecosystem.

- This PR is meant to give devenv users a quick way to have a `"virtualenv"` aware PS1, and they can customize it further by providing the Starship TOML with their project. Honestly, originally I just wanted to add `starship.enable` but, since Starship's logic for the prompt is too naive at times, it is useless/frustating to provide it without a way to change the prompt options.